app/src/test/java/org/mozilla/fenix/settings/creditcards/CreditCardsSettingFragmentTest.kt

```diff
@@ -0,0 +1,69 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.creditcards
+
+import androidx.fragment.app.FragmentActivity
+import androidx.preference.Preference
+import io.mockk.mockk
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestCoroutineDispatcher
+import mozilla.components.concept.storage.CreditCard
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.getPreferenceKey
+import org.mozilla.fenix.helpers.FenixRobolectricTestRunner
+import org.robolectric.Robolectric
+
+@ExperimentalCoroutinesApi
+@RunWith(FenixRobolectricTestRunner::class)
+class CreditCardsSettingFragmentTest {
+
+    private val testDispatcher = TestCoroutineDispatcher()
+    private lateinit var creditCardsSettingFragment: CreditCardsSettingFragment
+
+    @Before
+    fun setUp() {
+        creditCardsSettingFragment = CreditCardsSettingFragment()
+        val activity = Robolectric.buildActivity(FragmentActivity::class.java).create().get()
+
+        activity.supportFragmentManager.beginTransaction()
+            .add(creditCardsSettingFragment, "CreditCardsSettingFragmentTest")
+            .commitNow()
+        testDispatcher.advanceUntilIdle()
+    }
+
+    @Test
+    fun `manage credit cards pref is visible if there are saved cards`() {
```

Comment:
Should we use GIVEN WHEN THEN for these tests?

```suggestion
    fun `GIVEN a list of saved credit cards THEN the "Manage Credit Card" preference is visible`() {
```

Reply:
Good ideea!